Components: Dismiss icon button tooltip when clicked

aduth opened the pull request:
This pull request seeks to revise the behavior of the IconButton to dismiss its visible tooltip after being clicked / pressed. This is intended to eliminate visual clutter under the assumption that the tooltip provides value only prior to making the decision to click the button. The dismissal state is reset when either the cursor or focus leaves the button.

__Testing instructions:__

1. Navigate to Gutenberg > Demo
2. Select a block
3. Hover the Move Down button until the tooltip becomes visible
4. Click the Move Down button
5. Note that the tooltip is dismissed
6. Move mouse outside the button
7. Repeat step 3, noting that the tooltip becomes visible once more

youknowriad commented on components/icon-button/index.js:
```diff
@@ -16,21 +16,56 @@ import Tooltip from '../tooltip';
+		};
+	}
+
+	dismissTooltip() {
```
When I first looked at the changed, I was wondering where this prop was being handled because when you read

`onClick={ this.dismissTooltip }` you do not assume this would call the onClick prop. I wonder if it's a good idea to rename this function maybe just `onClick`

youknowriad commented:
I'm seeing a small "bug" when using the inserter or the block's menu (button with dropdown). 

Try these steps:

 - Click the button to open the inserter
 - Move the mouse slightly but keep it on the button
 - Click the button again
 - The tooltip flickers.

Not sure if this is easily fixable.

aduth commented:
There seems to be some issues with this conflicting with focus return too. Pressing escape from the opened inserter doesn't reset focus back to the button.

aduth committed:
Components: Always invoke original event handler

Regardless of whether tooltip logic is bypassed (disabled, no change), we should still emit the event to the original child

aduth committed:
Components: Dismiss Tooltip by click

aduth commented:
In 85e5dbe I opted for a much simpler approach, which is to bind to the click event from Tooltip component, treating it much the same way as a mouse leave event. This also means the behavior applies to any Tooltip'd element, not just IconButton.

aduth commented:
Reference to original approach, which is to be dropped in a rebase: c3a1e5e4a95313ea8a9d075edbaeb06d045b074c
